api/v1beta1/v1beta2_condition_consts.go

```diff
@@ -0,0 +1,262 @@
+	ClusterTopologyReconciledV1Beta2Condition = "TopologyReconciled"
+
+	// ClusterInfrastructureReadyV1Beta2Condition Mirror of Cluster's infrastructure Ready condition.
+	ClusterInfrastructureReadyV1Beta2Condition = InfrastructureReadyV1beta2Condition
+
+	// ClusterControlPlaneInitializedV1Beta2Condition is true when the Cluster's control plane is functional enough
+	// to accept requests. This information is usually used as a signal for starting all the provisioning operations
+	// that depends on a functional API server, but do not require a full HA control plane to exists.
+	ClusterControlPlaneInitializedV1Beta2Condition = "ControlPlaneInitialized"
+
+	// ClusterControlPlaneAvailableV1Beta2Condition is a mirror of Cluster's control plane Available condition.
+	ClusterControlPlaneAvailableV1Beta2Condition = "ControlPlaneAvailable"
+
+	// ClusterWorkersAvailableV1Beta2Condition is the summary of MachineDeployment and MachinePool's Available conditions.
+	ClusterWorkersAvailableV1Beta2Condition = "WorkersAvailable"
+
+	// ClusterMachinesReadyV1Beta2Condition surfaces detail of issues on the controlled machines, if any.
+	ClusterMachinesReadyV1Beta2Condition = MachinesReadyV1beta2Condition
+
+	// ClusterMachinesUpToDateV1Beta2Condition surfaces details of Cluster's machines not up to date, if any.
+	ClusterMachinesUpToDateV1Beta2Condition = MachinesUpToDateV1beta2Condition
+
+	// ClusterRemoteConnectionProbeV1Beta2Condition is true when control plane can be reached; in case of connection problems.
+	// The condition turns to false only if the cluster cannot be reached for 50s after the first connection problem
+	// is detected (or whatever period is defined in the --remote-connection-grace-period flag).
+	ClusterRemoteConnectionProbeV1Beta2Condition = "RemoteConnectionProbe"
+
+	// ClusterScalingUpV1Beta2Condition is true if available replicas < desired replicas.
+	ClusterScalingUpV1Beta2Condition = ScalingUpV1beta2Condition
+
+	// ClusterScalingDownV1Beta2Condition is true if replicas > desired replicas.
+	ClusterScalingDownV1Beta2Condition = ScalingDownV1beta2Condition
+
+	// ClusterRemediatingV1Beta2Condition surfaces details about ongoing remediation of the controlled machines, if any.
+	ClusterRemediatingV1Beta2Condition = RemediatingV1beta2Condition
+
+	// ClusterDeletingV1Beta2Condition surfaces details about ongoing deletion of the cluster.
+	ClusterDeletingV1Beta2Condition = DeletingV1beta2Condition
+
+	// ClusterPausedV1Beta2Condition is true if Cluster and all the resources being part of it are paused.
+	ClusterPausedV1Beta2Condition = PausedV1beta2Condition
+)
+
+// Conditions that will be used for the MachineHealthCheck object in v1Beta2 API version.
+const (
+	// MachineHealthCheckReadyV1beta2Condition is true if the MachineHealthCheck is not deleted,
+	// and RemediationAllowed condition is true.
+	MachineHealthCheckReadyV1beta2Condition = ReadyV1beta2Condition
```

Comment:
Really not sure about this one, can we maybe keep this out of scope?

Especially as we were planning to focus MHC on pure healthchecking, having the remediation decisions elsewhere + we deprecated fields accordingly


Also the proposal was making a point that we don't need Ready conditions on all objects

(IIRC Ready doesn't exist for MHCs today, please correct me if I'm wrong)